src/context/AppContext.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+import { SortType } from '../enums/SortType';
+import { getTodos } from '../api/todos';
+
+type AppContextContainerProps = {
+  error: string | null;
+  handleClickCloseError: () => void;
+  filterType: SortType;
+  changeFilterType: (newType: SortType) => void;
+  inputRef: React.RefObject<HTMLInputElement>;
+  todos: Todo[] | null;
+};
+
+type Props = {
+  children: ReactNode;
+};
+
+const AppContextContainer = createContext({} as AppContextContainerProps);
+
+export const useAppContextContainer = () => {
+  return useContext(AppContextContainer);
+};
```

Comment:
It would be also good to throw error in case when useContext returns undefined which means that the component that tries to use this hook is not wrapped inside context. Its then easier to debug errors.